### PR TITLE
DEVELOP-1305-2: Updated empty states for extensions: gitlab

### PR DESCRIPTION
**Pull request by jemmyw**
Add new empty state component and updated menus

CleanShot 2022-08-11 at 14 34 52@2x

CleanShot 2022-08-11 at 14 37 00@2x

Requires https://github.com/aha-app/aha-app/pull/23132 for the icon, but this is not a release blocker

### CODE REVIEW

File: src/lib/extension.ts [hunk 1]
```diff
@@ -1 +1,3 @@
 export const IDENTIFIER = 'aha-develop.gitlab';
+export const LEARN_MORE_URL = 'https://github.com/aha-develop/gitlab';
+export const ICON = aha.iconForExtensionIdentifier(IDENTIFIER);
```
Comment (trydionel) [inline src/lib/extension.ts]: Does this PR depend on an unreleased change to the `aha` global? I'm seeing this exception in the endpoint handler logs:

```
TypeError: aha.iconForExtensionIdentifier is not a function
JavaScript at src/lib/extension.ts:3:25
JavaScript at aha-develop.gitlab.webhook:239:3
JavaScript at aha-develop.gitlab.webhook:240:3
JavaScript at Aha.defineExtension (ContributionRegistry.js:50:7)
JavaScript at aha-develop.gitlab.webhook:1:34
JavaScript at aha-develop.gitlab.webhook:241:4
```

**Review by trydionel — CHANGES_REQUESTED** [review]

File: src/lib/extension.ts (hunk 1 above)
Comment (trydionel) [reply src/lib/extension.ts]: Oh just saw this in the PR message, duh: "Requires https://github.com/aha-app/aha-app/pull/23132 for the icon, but this is not a release blocker"

Based on the behavior in the endpoint contribution, I believe this is a blocker though: the handler fails to run due to this exception.

Comment (jemmyw) [reply src/lib/extension.ts]: ah, I understand. I need to not include that call in the endpoint

Comment (jemmyw) [reply src/lib/extension.ts]: @trydionel this should be fixed now

File: src/components/attribute/EmptyState.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  const [hasConfiguredWebhook, setHasConfiguredWebhook] = useState<Boolean | null>(null);
+
+  // aha-menu gets mad if you remove it from the DOM while the menu is open.
+  // Wait 1 tick so the menu has a chance to unmount first.
+  const viewPasteMode = () => {
+    setTimeout(() => setPasteMode(true), 1);
+  };
+
+  const pasteLink = async (mrUrl: string) => {
+    if (!validateMrURL(mrUrl)) {
```
Comment (louisfnz) [inline src/components/attribute/EmptyState.tsx:60]: If I put in a junk string that's not a URL, the `new URL` constructor will throw, so `validation` doesn't get set. Might be worth a `try/catch` in `validateMrURL()`?

**Review by louisfnz — APPROVED** [review]

**Pull request merged**